### PR TITLE
RTPS 2.4 Removed unused PID’s; Adjusted RELIABLE PSM

### PR DESCRIPTION
**Pull request by lij-oci**
RTPS 2.4 Removed unused PID’s; Adjusted RELIABLE PSM mapping

### CODE REVIEW

File: dds/DCPS/RTPS/ParameterListConverter.cpp [hunk 1]
```diff
@@ -1059,9 +956,15 @@ int from_param_list(const ParameterList& param_list,
         writer_data.ddsPublicationData.reliability = param.reliability();
         // Interoperability note:
         // Spec creators for RTPS have reliability indexed at 1
-        writer_data.ddsPublicationData.reliability.kind =
-          (DDS::ReliabilityQosPolicyKind)
-              ((int)writer_data.ddsPublicationData.reliability.kind - 1);
+        {
+          const CORBA::Short rtpsKind = (const CORBA::Short)(param.reliability().kind);
+          if(rtpsKind == RTPS::BEST_EFFORT){
```
Comment (simpsont-oci) [inline dds/DCPS/RTPS/ParameterListConverter.cpp]: Ditto above... fix the whitespace / braces to match the guidelines. Two more instances below.

File: dds/DCPS/RTPS/ParameterListConverter.cpp [hunk 2]
```diff
@@ -507,15 +456,6 @@ int to_param_list(const ParticipantProxy_t& proxy,
     add_param(param_list, eiq_param);
   }
 
-  Parameter abe_param;
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/ParameterListConverter.cpp]: Let's not remove this part (lines 510-518).

File: dds/DCPS/RTPS/ParameterListConverter.cpp [hunk 3]
```diff
@@ -593,15 +528,7 @@ int from_param_list(const ParameterList& param_list,
         proxy.expectsInlineQos =
             param.expects_inline_qos();
         break;
-      case PID_PARTICIPANT_BUILTIN_ENDPOINTS:
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/ParameterListConverter.cpp]: Also, keep this (lines 596-604).

**Comment by lij-oci** [comment]
DirectedWrite still does not work when the subscriber is set to Best Effort.
[subscriber_BestEffort.log](https://github.com/objectcomputing/OpenDDS/files/3871999/subscriber_BestEffort.log)
[subscriber_Reliable.log](https://github.com/objectcomputing/OpenDDS/files/3872000/subscriber_Reliable.log)

**Pull request merged**